routes/nft.go

```diff
@@ -1338,7 +1340,389 @@ func (fes *APIServer) _bidEntryToResponse(bidEntry *lib.NFTBidEntry, postEntryRe
+	NFTPostHashHex              string `safeForLogging:"true"`
+	SerialNumber                int    `safeForLogging:"true"`
+
+	MinFeeRateNanosPerKB uint64 `safeForLogging:"true"`
+}
+
+type BurnNFTResponse struct {
+	UpdaterPublicKeyBase58Check string `safeForLogging:"true"`
+	NFTPostHashHex              string `safeForLogging:"true"`
+	SerialNumber                int    `safeForLogging:"true"`
+
+	TotalInputNanos   uint64
+	ChangeAmountNanos uint64
+	FeeNanos          uint64
+	Transaction       *lib.MsgBitCloutTxn
+	TransactionHex    string
+}
+
+func (fes *APIServer) BurnNFT(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := BurnNFTRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("BurnNFT: Error parsing request body: %v", err))
+		return
+	}
+
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("BurnNFT: Error getting utxoView: %v", err))
+		return
+	}
+
+	// Do a simple validation of the requestData.
+	if requestData.NFTPostHashHex == "" {
+		_AddBadRequestError(ww, fmt.Sprintf("BurnNFT: Must include NFTPostHashHex"))
+		return
+
+	} else if requestData.UpdaterPublicKeyBase58Check == "" {
+		_AddBadRequestError(ww, fmt.Sprintf("BurnNFT: Must include UpdaterPublicKeyBase58Check"))
+		return
+
+	} else if requestData.SerialNumber <= 0 || requestData.SerialNumber > int(utxoView.GlobalParamsEntry.MaxCopiesPerNFT) {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"BurnNFT: SerialNumbers must be between %d and %d, received: %d",
+			1, utxoView.GlobalParamsEntry.MaxCopiesPerNFT, requestData.SerialNumber))
+		return
+
+	}
+
+	// Get the PostHash for the NFT.
+	nftPostHashBytes, err := hex.DecodeString(requestData.NFTPostHashHex)
+	if err != nil || len(nftPostHashBytes) != lib.HashSizeBytes {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"BurnNFT: Error parsing post hash %v: %v",
+			requestData.NFTPostHashHex, err))
+		return
+	}
+	nftPostHash := &lib.BlockHash{}
+	copy(nftPostHash[:], nftPostHashBytes)
+
+	// Get the updater's public key.
+	updaterPublicKeyBytes, _, err := lib.Base58CheckDecode(requestData.UpdaterPublicKeyBase58Check)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("BurnNFT: Problem decoding updater public key: %v", err))
+		return
+	}
+
+	// Get the NFT in question so we can do a more hardcore validation of the request data.
+	nftKey := lib.MakeNFTKey(nftPostHash, uint64(requestData.SerialNumber))
+	nftEntry := utxoView.GetNFTEntryForNFTKey(&nftKey)
+	if nftEntry == nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"BurnNFT: Error could not find the NFT an NFT with postHash %v and serialNumber %d",
+			requestData.NFTPostHashHex, requestData.SerialNumber))
+		return
+
+	} else if nftEntry.IsForSale {
+		_AddBadRequestError(ww, fmt.Sprintf("BurnNFT: Cannot burn an NFT that is for sale."))
+		return
+	}
+
```

Comment:
Should we validate that the updater is currently the owner of this NFT?

Reply:
Added and tested :+1: